src/fidesops/analytics.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from datetime import datetime, timezone
+from platform import system
+
+import versioneer
+from fideslog.sdk.python.client import AnalyticsClient
+from fideslog.sdk.python.event import AnalyticsEvent
+from fideslog.sdk.python.exceptions import AnalyticsError
+from fideslog.sdk.python.utils import generate_client_id
+
+from fidesops.core.config import config
+from fidesops.main import app
+from fidesops.schemas.analytics import FideslogAnalyticsEvent
+
+logger = logging.getLogger(__name__)
+
+
+def get_version() -> str:
+    """Version of Fidesops"""
+    return versioneer.get_version()
+
+
+def in_developer_mode() -> bool:
+    """True if runnning in dev mode"""
+    return config.dev_mode
+
+
+def in_docker_container() -> bool:
+    """`True` if the command was submitted within a Docker container. Default: `False`."""
+    return True
+
+
+def running_on_local_host() -> bool:
+    """For events submitted as a result of making API server requests, `True` if the API server is running on the user's local host. Default: `False`."""
+    return False
+
+
+class Analytics:
```

Comment:
I'll reiterate that I don't think this class is necessary, and that you're creating a lot of undue complexity with it. The fideslog SDK is designed to be small, strict, and intuitive. Rather than adding this class + the `FideslogAnalyticsEvent` schema, I think you'll be fine with just:
```python
from fideslog.sdk.python.client import AnalyticsClient
from fideslog.sdk.python.utils import FIDESOPS, generate_client_id

analytics_client = AnalyticsClient(
    client_id=config.user.analytics_id or generate_client_id(FIDESOPS),
    ...
)
```
Then, throughout the fidesops codebase, you'll be able to do:
```python
from fideslog.sdk.python.event import AnalyticsEvent
from fideslog.sdk.python.exceptions import AnalyticsError
from fidesops.analytics import analytics_client

if not config.user.analytics_opt_out:
    try:
        analytics_client.send(AnalyticsEvent(...))
    except AnalyticsError as err:
        logger.warning(f"Failed to send analytics event: {err}")
```

Reply:
I'd prefer to handle checking for `config.user.analytics_opt_out` and exceptions at the analytics class level, so that we don't duplicate code. We may also need to pass in user id in the future for retrieving analytics permissions which I'd also like to keep scoped to the analytics class.

Reply:
My biggest unknown, though, is  `generate_ client_id`. I know you said this is supposed to be unique per client, but you mentioned in our screenshare that I should instead make it unique per user + instance. I guess I just need more clarity around this, if fidesops requires sending user-level event data.